packages/legend-application-studio/src/stores/editor-state/element-editor-state/service/ServiceRegistrationState.ts

```diff
@@ -314,5 +319,34 @@ export class ServiceRegistrationState {
         'Service version can not be empty in Semi-interactive and Prod service type',
       );
     }
+
+    // validate service parameter multiplicity. Currently [1..*] is not allowed for service registration.
+    const queryLambda = (this.service.execution as PureExecution).func;
+    const parameters = (queryLambda.parameters ?? []) as object[];
+    if (parameters.length) {
+      const params = buildLambdaVariableExpressions(
+        queryLambda,
+        this.editorStore.graphManagerState,
+      )
+        .map((p) =>
+          observe_ValueSpecification(
+            p,
+            this.editorStore.changeDetectionState.observerContext,
+          ),
+        )
+        .filter(filterByType(VariableExpression))
+        .filter(
+          (p) =>
+            p.multiplicity.lowerBound === 1 &&
+            p.multiplicity.upperBound === undefined,
+        );
+      if (params.length > 0) {
+        this.editorStore.applicationStore.notifyError(
```

Comment:
you don't need to notify the error in this function, all you need to do is throw. The function call catches this error and notifies the user.